### PR TITLE
Np 16238 adding query parameters cristin

### PR DESCRIPTION
**Pull request by tkardu**
I am still not sure if it is correct this way. Putting all those in the Json properties (for future use?)

### CODE REVIEW

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/common/CristinQueryTest.java
Comment (torarnet) [inline cristin-project/src/test/java/no/unit/nva/cristin/projects/common/CristinQueryTest.java:56]: these 3 methods (`buildReturnsUriWithExtendedListOfParameters, buildReturnsUriWithKeyword and buildReturnsUriApproved`) can be combined into one, matching against a uri string with all parameters.

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/common/CristinQueryTest.java
Comment (torarnet) [inline cristin-project/src/test/java/no/unit/nva/cristin/projects/common/CristinQueryTest.java:23]: very short names

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/query/projectQuery.feature
Comment (torarnet) [inline cristin-project/src/test/java/no/unit/nva/cristin/projects/query/projectQuery.feature:103]: I did not see this until now, but maybe we can consider using " as it was instead of ' to avoid escaping ' in the string itself. But its not a must and we can have it as is

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/query/QueryCristinProjectHandlerTest.java
Comment (torarnet) [inline cristin-project/src/test/java/no/unit/nva/cristin/projects/query/QueryCristinProjectHandlerTest.java:709]: consider extracting `captor.getValue().toString()` into a local variable so we dont have to type the whole chain each time

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/query/QueryCristinProjectHandlerTest.java
Comment (torarnet) [inline cristin-project/src/test/java/no/unit/nva/cristin/projects/query/QueryCristinProjectHandlerTest.java:648]: Why do you not want to use static imports? There is alot of changes in this class just reverting already existing code because of not using static imports. When i first started in this team i was told the same, that using static imports is better. Maybe I need a second opinion from another reviewer.

File: cristin-commons/src/main/java/no/unit/nva/cristin/model/JsonPropertyNames.java
Comment (torarnet) [inline cristin-commons/src/main/java/no/unit/nva/cristin/model/JsonPropertyNames.java:53]: Maybe this should have been refactored in another PR (this is some old code already present, not something you introduced) to use CristinQuery class or another new query param constants class as these values are not json field values but query parameters.

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/query/QueryCristinProjectHandler.java
Comment (torarnet) [inline cristin-project/src/main/java/no/unit/nva/cristin/projects/query/QueryCristinProjectHandler.java:2]: I like the code you have written in this class. Very creative and reduces a lot of one by one parameter logic 🙂 You can also consider moving the logic shared between this handler class and the other handler class to avoid duplication. We can discuss that later, as im not sure where we can put it, but my first thought is either their superclass and/or cristin-commons module's util class

**Review by torarnet — COMMENTED** [review]
See comments. You should also update openapi in docs folder so we know what query parameters we support. 

Overall, mostly minor issues and also some good additions done by you 🙂 . Please see other reviewers comments when he has time before doing too many new changes.

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/query/QueryCristinProjectHandlerTest.java
Comment (tkardu) [reply cristin-project/src/test/java/no/unit/nva/cristin/projects/query/QueryCristinProjectHandlerTest.java:648]: I was just trying to avoid this issue of the * imports.

File: cristin-commons/src/main/java/no/unit/nva/cristin/model/JsonPropertyNames.java
Comment (tkardu) [reply cristin-commons/src/main/java/no/unit/nva/cristin/model/JsonPropertyNames.java:53]: I was slightly confused with that too.

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/common/CristinQueryTest.java
Comment (tkardu) [reply cristin-project/src/test/java/no/unit/nva/cristin/projects/common/CristinQueryTest.java:21]: I do not agree. As stands in documentation: _Cristin id, acronym, name, or part of the name of the coordinating institution_

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/common/CristinQueryTest.java
Comment (tkardu) [reply cristin-project/src/test/java/no/unit/nva/cristin/projects/common/CristinQueryTest.java:23]: In documentation it says you can just use "part of the name". I used those because they provided enough results in real cristin queries so I can check real combination of sample params. But if want them longer, easy to fix

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/common/CristinQueryTest.java
Comment (torarnet) [reply cristin-project/src/test/java/no/unit/nva/cristin/projects/common/CristinQueryTest.java:21]: You are correct, marking as resolved.

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/common/CristinQueryTest.java
Comment (torarnet) [reply cristin-project/src/test/java/no/unit/nva/cristin/projects/common/CristinQueryTest.java:23]: Just keep them as they are

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/query/QueryCristinProjectHandlerTest.java
Comment (torarnet) [reply cristin-project/src/test/java/no/unit/nva/cristin/projects/query/QueryCristinProjectHandlerTest.java:648]: * imports is not used for static imports. a static import can be `import static no.unit.nva.model.Organization.ORGANIZATION_CONTEXT;`

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/common/CristinQuery.java
Comment (torarnet) [inline cristin-project/src/main/java/no/unit/nva/cristin/projects/common/CristinQuery.java:268]: you forgot to add this in `generateQueryParameters`

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/common/CristinQueryTest.java
Comment (tkardu) [reply cristin-project/src/test/java/no/unit/nva/cristin/projects/common/CristinQueryTest.java:54]: Merged tests

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/common/CristinQuery.java
Comment (torarnet) [inline cristin-project/src/main/java/no/unit/nva/cristin/projects/common/CristinQuery.java:309]: use static import

**Review by torarnet — APPROVED** [review]

**Pull request merged**
